cpmpy/tools/dimacs.py

```diff
@@ -72,41 +72,43 @@ def write_dimacs(model, fname=None):
 def read_dimacs(fname):
     """
         Read a CPMpy model from a DIMACS formatted file
-        If the number of variables and constraints is not present in the header, they are inferred.
         :param: fname: the name of the DIMACS file
         :param: sep: optional, separator used in the DIMACS file, will try to infer if None
     """
 
     m = cp.Model()
 
     with open(fname, "r") as f:
-
-        lines = f.readlines()
-        for i, line in enumerate(lines): # DIMACS allows for comments, skip comment lines
-            if line.startswith("p cnf"):
-                break
+        typ = None  # CNF/WCNF
+        # TODO infer p-header values (although generally not a good idea)
+        nr_cls = None
+        bvs = None
+        clause = []
+        for line in f.readlines():
+            if line.startswith("p"):
+                try:
+                    typ,typ,nr_vars,nr_cls = line.strip().split(" ")
```

Comment:
This already assumes the first two arguments are the same. Better to make this a separate check:
```
t1, t2, nr_vars, nc_cls = line.strip().split(" ")
assert t1 == t2 == "cnf", "Header line should be formed as "cnf cnf <#vars> <#clauses>"
```